homeassistant/components/xiaomi_miio/humidifier.py

```diff
@@ -0,0 +1,462 @@
+
+        if model in MODELS_HUMIDIFIER_MIOT:
+            air_humidifier = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+            entity = XiaomiAirHumidifierMiot(
+                name, air_humidifier, config_entry, unique_id
+            )
+        elif model.startswith("zhimi.humidifier."):
+            air_humidifier = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+            entity = XiaomiAirHumidifier(name, air_humidifier, config_entry, unique_id)
+        else:
+            _LOGGER.error(
+                "Unsupported humidifier device found! Please create an issue at "
+                "https://github.com/syssi/xiaomi_airpurifier/issues "
+                "and provide the following data: %s",
+                model,
+            )
+            return
+
+        hass.data[DATA_KEY][host] = entity
+        entities.append(entity)
+
+        async def async_service_handler(service):
+            """Map services to methods."""
+            method = SERVICE_TO_METHOD[service.service]
+            params = {
+                key: value
+                for key, value in service.data.items()
+                if key != ATTR_ENTITY_ID
+            }
+            entity_ids = service.data.get(ATTR_ENTITY_ID)
+            if entity_ids:
+                entities = [
+                    entity
+                    for entity in hass.data[DATA_KEY].values()
+                    if entity.entity_id in entity_ids
+                ]
+            else:
+                entities = hass.data[DATA_KEY].values()
+
+            update_tasks = []
+
+            for entity in entities:
+                entity_method = getattr(entity, method["method"], None)
+                if not entity_method:
+                    continue
+                await entity_method(**params)
+                update_tasks.append(
+                    hass.async_create_task(entity.async_update_ha_state(True))
+                )
+
+            if update_tasks:
+                await asyncio.wait(update_tasks)
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class XiaomiGenericHumidifierDevice(XiaomiMiioEntity, HumidifierEntity):
+    """Representation of a generic Xiaomi device."""
+
+    def __init__(self, name, device, entry, unique_id):
+        """Initialize the generic Xiaomi device."""
+        super().__init__(name, device, entry, unique_id)
+
+        self._available = False
+        self._state = None
+        self._state_attrs = {ATTR_MODEL: self._model}
+        self._skip_update = False
+        self._available_modes = []
+        self._mode = None
+        self._available_attributes = AVAILABLE_ATTRIBUTES
+        self._min_humidity = DEFAULT_MIN_HUMIDITY
+        self._max_humidity = DEFAULT_MAX_HUMIDITY
+        self._humidity_steps = 100
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return DEVICE_CLASS_HUMIDIFIER
```

Comment:
Prefer entity class attributes over properties for static values: https://developers.home-assistant.io/docs/core/entity#entity-class-attributes